lib/Config.js

```diff
@@ -280,6 +280,46 @@ function requestsConfigAssert(requestsConfig) {
     }
 }
 
+function bucketNotifAssert(bucketNotifConfig) {
+    assert(Array.isArray(bucketNotifConfig),
+        'bad config: bucket notification configuration must be an array');
+    bucketNotifConfig.forEach(c => {
+        const { resource, type, host, port, auth } = c;
+        assert(typeof resource === 'string',
+            'bad config: bucket notification configuration resource must be a string');
+        assert(typeof type === 'string',
+            'bad config: bucket notification configuration type must be a string');
+        assert(typeof host === 'string' && host !== '',
+            'bad config: hostname must be a non-empty string');
+        assert(Number.isInteger(port) && port > 0,
+            'bad config: port must be a positive integer');
+        assert(typeof auth === 'object',
+            'bad config: bucket notification auth must be an object');
+        const { cert, user, password } = auth;
+        if ((cert && user && password) || (!cert && !user && !password)) {
+            throw new Error('bad config: bucket notification configuration ' +
+                'auth should contain either cert or user and password');
+        }
+        if (cert) {
+            const certpath = (cert[0] === '/') ? cert : `${this._basePath}/${cert}`;
+            assert.doesNotThrow(() =>
```

Comment:
This assert doesn't feel necessary.
We try and do something and assert it doesn't fail, but when it does we throw an exception cause the assert failed.
It would be just as reasonable to just do accessSync and on failure, it will throw anyway.